src/Components/HandleDialog.js

```diff
@@ -0,0 +1,173 @@
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { logout } from "./Firebase";
+import { LocalUserContext } from "./LocalUserContext";
+import * as Yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useForm } from "react-hook-form";
+import {
+  SaveHandle,
+  CheckForHandleDuplicates,
+  SaveToFirestore,
+} from "./Firestore";
+
+export default function HandleDialog({ user }) {
+  const navigate = useNavigate();
+  const [localUser, setLocalUser] = useContext(LocalUserContext);
+  let newLocalUser;
+
+  const [open, setOpen] = useState(true);
+
+  const [handle, setHandle] = useState("");
+
+  const handleClose = (event, reason) => {
+    if (reason && reason === "backdropClick") return;
+    setOpen(false);
+  };
+
+  async function handleCancel() {
+    await logout();
+    setLocalUser({
+      name: [],
+      likes: [],
+      dislikes: [],
+      lists: [],
+      savedLists: [],
+      avatar: null,
+      bio: null,
+      top8: [],
+      reviews: [],
+      comments: [],
+      null: null,
+    });
+    navigate("/");
+  }
+
+  function handleFormSubmission() {
+    let isDuplicate;
+    newLocalUser = { ...localUser };
+    newLocalUser.handle = handle;
+    CheckForHandleDuplicates(handle, user)
+      .then((result) => {
```

Comment:
style note: since functions that return a promise (including 'then' functions) provide their return values to the next function as the argument, we could use that mechanism to provide `isDuplicate`, there is no need for the external local variable.  something like:
```js
CheckForHandleDuplicates(handle, user)
.then((isDuplicate) => {
  if (isDuplicate) {
    // oh no!
  } else {
    // yay!
  }
});
```

Reply:
Oooweeee that's clean!!! Thanks for the correction.